cookbooks/creating_a_behat_extension.rst

```diff
@@ -0,0 +1,270 @@
+Creating a Behat extension
+==========================
+
+Extensions are particularly useful when configuration becomes a necessity.
+
+In this cookbook, we will create a simple extension named ``LogsExtension`` that logs scenario durations and cover:
+
+#. Setting up the context
+#. Creating the extension
+#. Initializing the context
+
+Setting Up the Context
+----------------------
+
+First, we need to create a context that will handle the logging.
```

Comment:
I wouldn't necessarily say you *need* to implement a context if you only want to listen to events. It's not wrong,
and it works well in your example, but there are other solutions that might be better sometimes.

Maybe something like this?

```suggestion
First, we need to create a class that will handle the logging. There are several ways to do this (you can hook
directly into behat's event dispatcher, for example).

For our case, we will create a `Context` and use the :doc:`normal tagged hooks</user_guide/context/hooks>`
that you might have used in your own contexts.

This is a straightforward approach, and means we can also show how to initialise any context with custom
configuration or dependencies.
```